### PR TITLE
feat: Support clickhouse-logger plugin log option of request/response body

### PR DESCRIPTION
**Pull request by pixeldin**
### Description

Providing some optional setting items for clickhouse-logger to control the recording log content as requested in https://github.com/apache/apisix/issues/8670.

I refer some options of the [http-logger](https://apisix.apache.org/docs/apisix/next/plugins/http-logger/):
`include_req_body`, `include_req_body_expr`, `include_resp_body` and `include_resp_body_expr`

Fixes #([8670](https://github.com/apache/apisix/issues/8670))

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/plugin/clickhouse-logger2.t [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+--- response_body
+passed
+
+
+
+=== TEST 2: collect response log
+--- request
+GET /hello
+--- error_log eval
+qr/\{.*"body":"hello world\\n"/
```
Comment (spacewander) [inline t/plugin/clickhouse-logger2.t:81]: Could you also include the `clickhouse body: ` prefix? So people can know this output is from the dummy upstream, not from somewhere else.

File: t/plugin/clickhouse-logger2.t [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+--- response_body
+passed
+
+
+
+=== TEST 4: skip collect response log for condition
+--- request
+GET /hello?foo=unknown
+--- no_error_log
+"response":{"body":"hello world\n"
```
Comment (spacewander) [inline t/plugin/clickhouse-logger2.t:137]: Ditto

**Comment by gaoxingliang** [comment]
@pixeldin Thanks for your work. I will upgrade to this version when it's released.

**Comment by gaoxingliang** [comment]
Hi @pixeldin how can I avoid recording full response body if it's too long? (Can I truncate the response to max 1024 bytes?)

File: t/plugin/clickhouse-logger2.t (hunk 1 above)
Comment (pixeldin) [reply t/plugin/clickhouse-logger2.t:81]: lgtm, I will cover it.

File: t/plugin/clickhouse-logger2.t (hunk 2 above)
Comment (pixeldin) [reply t/plugin/clickhouse-logger2.t:137]: ok, I will cover it.

**Comment by pixeldin** [comment]
> Hi @pixeldin how can I avoid recording full response body if it's too long? (Can I truncate the response to max 1024 bytes?)

IMO, truncating body by length may lose some data unexpected. What about trying to extract the target response body you concerned in upstream or middle layer at first?

**Comment by gaoxingliang** [comment]
> > Hi @pixeldin how can I avoid recording full response body if it's too long? (Can I truncate the response to max 1024 bytes?)
> 
> IMO, truncating body by length may lose some data unexpected. What about trying to extract the target response body you concerned in upstream or middle layer at first?

I'm trying to record the request/response into clickhouse for audit purpose. while the json response maybe too large.  I think if we have some normal / quick config, then I don't need to write code to do that. (For now, I set a column `response` to store all response related data.) I don't think it's a good idea if the body in response is too large when doing olap.

File: t/plugin/clickhouse-logger2.t [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+use t::APISIX 'no_plan';
+
+log_level("info");
+#repeat_each(1);
```
Comment (spacewander) [inline t/plugin/clickhouse-logger2.t:21]: We can remove it?

**Review by spacewander — DISMISSED** [review]

**Comment by pixeldin** [comment]
> > > Hi @pixeldin how can I avoid recording full response body if it's too long? (Can I truncate the response to max 1024 bytes?)
> > 
> > 
> > IMO, truncating body by length may lose some data unexpected. What about trying to extract the target response body you concerned in upstream or middle layer at first?
> 
> I'm trying to record the request/response into clickhouse for audit purpose. while the json response maybe too large. I think if we have some normal / quick config, then I don't need to write code to do that. (For now, I set a column `response` to store all response related data.) I don't think it's a good idea if the body in response is too large when doing olap. I also checked the operators here - https://github.com/api7/lua-resty-expr#operator-list, It doesn't provide a operator like `length`

I know your mean. As I know, there is no suggestion for me about the body length limit in this scenario. Maybe you can start and create a new issue for this.

**Review by spacewander — APPROVED** [review]

**Comment by soulbird** [comment]
Please resolve the conflicts @pixeldin

**Comment by spacewander** [comment]
Let's merge master to solve the conflict.

**Review by soulbird — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**